src/UserManager.ts

```diff
@@ -323,30 +323,35 @@ export class UserManager {
 
     public async signinCallback(url = window.location.href): Promise<User | null> {
         const { state } = await this._client.readSigninResponseState(url);
-        if (state.request_type === "si:r") {
-            return this.signinRedirectCallback(url);
+        switch (state.request_type) {
+            case "si:r":
+                return this.signinRedirectCallback(url);
+            case "si:p":
+                await this.signinPopupCallback(url);
+                return null;
```

Comment:
I think the `null` return value on this function is a bit strange given that all of the other `*Callback` methods use `void` as their return type. Could we update this function to have a `Promise<User | void>` return value and simplify these to 

```suggestion
                return await this.signinPopupCallback(url);
```
?

Reply:
hmm `signinPopupCallback` calls `this._popupNavigator.callback`, which returns void. I have looked into the callback method, no glue were to get the user from...
Any way this is something else than what this merge request fixes. Can you provide an improvement for that?

Reply:
Yeah there's no way to get a user from the popup/iframe callbacks, the `void` return is appropriate. I only figured this change is relevant here since you're making tests that enforce returning `null`.

Reply:
ahh got it `void` vs `null`. Will adapt...